Mathlib/Topology/Category/LightProfinite/Limits.lean

```diff
@@ -0,0 +1,254 @@
+    (fun _ => pullback.lift_fst _ _ _ _ _)
+    (fun _ => pullback.lift_snd _ _ _ _ _)
+    (fun _ _ hm => pullback.hom_ext _ _ _ _ (hm .left) (hm .right))
+
+end Pullback
+
+section FiniteCoproduct
+
+instance {α : Type w} [Finite α] (X : α → ProfiniteMax.{u, w}) [∀ a, (X a).IsLight] :
+    (Profinite.finiteCoproduct X).IsLight where
+  countable_clopens := by
+    refine @Function.Surjective.countable ((a : α) → Clopens (X a)) _ inferInstance
+      (fun f ↦ ⟨⋃ (a : α), Sigma.mk a '' (f a).1, ?_⟩) ?_
+    · apply isClopen_iUnion_of_finite
+      intro i
+      exact ⟨isClosedMap_sigmaMk _ (f i).2.1, isOpenMap_sigmaMk _ (f i).2.2⟩
+    · intro ⟨s, ⟨hsc, hso⟩⟩
+      rw [isOpen_sigma_iff] at hso
+      rw [isClosed_sigma_iff] at hsc
+      refine ⟨fun i ↦ ⟨_, ⟨hsc i, hso i⟩⟩, ?_⟩
+      simp only [Subtype.mk.injEq]
+      ext ⟨i, xi⟩
+      refine ⟨fun hx ↦ ?_, fun hx ↦ ?_⟩
+      · simp only [Clopens.coe_mk, Set.mem_iUnion] at hx
+        obtain ⟨_, _, hj, hxj⟩ := hx
+        simpa [hxj] using hj
+      · simp only [Clopens.coe_mk, Set.mem_iUnion]
+        refine ⟨i, xi, (by simpa using hx), rfl⟩
+
+variable {α : Type w} [Finite α] (X : α → LightProfiniteMax.{u, w})
+
+/--
+The "explicit" coproduct of a finite family of objects in `LightProfinite`, whose underlying
+profinite set is the disjoint union with its usual topology.
+-/
+noncomputable
+def finiteCoproduct : LightProfinite :=
+  ofIsLight (Profinite.finiteCoproduct.{u, w} fun a ↦ (X a).toProfinite)
+
+/-- The inclusion of one of the factors into the explicit finite coproduct. -/
+def finiteCoproduct.ι (a : α) : X a ⟶ finiteCoproduct X where
+  toFun := (⟨a, ·⟩)
+  continuous_toFun := continuous_sigmaMk (σ := fun a => (X a).toProfinite)
+
+/--
+To construct a morphism from the explicit finite coproduct, it suffices to
+specify a morphism from each of its factors. This is the universal property of the coproduct.
+-/
+def finiteCoproduct.desc {B : LightProfiniteMax.{u, w}} (e : (a : α) → (X a ⟶ B)) :
+    finiteCoproduct X ⟶ B where
+  toFun := fun ⟨a, x⟩ => e a x
+  continuous_toFun := by
+    apply continuous_sigma
+    intro a
+    exact (e a).continuous
+
+@[reassoc (attr := simp)]
+lemma finiteCoproduct.ι_desc {B : LightProfiniteMax.{u, w}} (e : (a : α) → (X a ⟶ B)) (a : α) :
+    finiteCoproduct.ι X a ≫ finiteCoproduct.desc X e = e a := rfl
+
+lemma finiteCoproduct.hom_ext {B : LightProfiniteMax.{u, w}} (f g : finiteCoproduct X ⟶ B)
+    (h : ∀ a : α, finiteCoproduct.ι X a ≫ f = finiteCoproduct.ι X a ≫ g) : f = g := by
+  ext ⟨a, x⟩
+  specialize h a
+  apply_fun (· x) at h
+  exact h
+
+/-- The coproduct cocone associated to the explicit finite coproduct. -/
+@[simps]
+noncomputable def finiteCoproduct.cofan : Limits.Cofan X where
+  pt := finiteCoproduct X
+  ι := Discrete.natTrans fun ⟨a⟩ => finiteCoproduct.ι X a
+
+/-- The explicit finite coproduct cocone is a colimit cocone. -/
+@[simps]
+def finiteCoproduct.isColimit : Limits.IsColimit (finiteCoproduct.cofan X) where
```

Comment:
Here, you should use `mkCofanColimit`. Then, the `s` you will get will be cofans instead of cocones, so that you may use the syntax `s.inj a` instead of `s.ι.app ⟨a⟩`.

Reply:
I took the opportunity to apply this suggestion and the previous one in `Profinite/Limits` and `CompHaus/Limits` as well